lib/lib.go

```diff
@@ -694,8 +691,11 @@ func (inst *Instance) Connect(ctx context.Context) (err error) {
 	// for now if we have an IPFS node instance, node.GoOnline has to make a new
 	// instance to connect properly. If remoteClient or remote retains the reference to the
 	// old instance, we run into issues where the online instance can't "see"
-	// the additions. We fix that by re-initializing the client and remote with the new
-	// instance
+	// the additions. We fix that by shutting down the previous client and
+	// re-initializing the client and remote with the new instance
+	if inst.remoteClient != nil {
+		<-inst.remoteClient.Shutdown()
+	}
```

Comment:
Added `Shutdown` method to `remote.Client`. This can also be refactored into lib, and have lib control the shutdown, but this felt cleaner. Would love opinions tho.